subprojects/common/core/src/main/java/hu/bme/mit/theta/core/utils/StmtToExprTransformer.java

```diff
@@ -146,12 +148,13 @@ public StmtUnfoldResult visit(NonDetStmt nonDetStmt, VarIndexing indexing) {
             final List<VarIndexing> indexings = new ArrayList<>();
             VarIndexing jointIndexing = indexing;
             int count = 0;
-            VarDecl<IntType> tempVar = VarPoolUtil.requestInt();
+            //            VarDecl<IntType> tempVar = VarPoolUtil.requestInt();
             for (Stmt stmt : nonDetStmt.getStmts()) {
-                final Expr<BoolType> tempExpr = Eq(
-                        ExprUtils.applyPrimes(tempVar.getRef(), indexing), Int(count++));
-                final StmtUnfoldResult result = toExpr(stmt, indexing.inc(tempVar));
-                choices.add(And(tempExpr, And(result.exprs)));
+                //                final Expr<BoolType> tempExpr = Eq(
+                //                        ExprUtils.applyPrimes(tempVar.getRef(), indexing),
+                // Int(count++));
+                final StmtUnfoldResult result = toExpr(stmt, indexing /*.inc(tempVar)*/);
+                choices.add(/*And(tempExpr, */ And(result.exprs) /*)*/);
                 indexings.add(result.indexing);
                 jointIndexing = jointIndexing.join(result.indexing);
             }
```

Comment:
Is this OK?

Reply:
Yes, this is intentional, comments can even be removed